src/components/services-view/services-view.controller.ts

```diff
@@ -2,178 +2,188 @@ import * as angular from 'angular';
-
-        return subCats;
-    };
-
-    public handleClick = (item: any, e: any) => {
-        this.ctrl.serviceToOrder = item;
-        this.ctrl.openOrderingPanel();
-    };
-
-    public openOrderingPanel() {
-        this.ctrl.orderingPanelvisible = true;
-    };
-
-    public closeOrderingPanel = () => {
-        this.ctrl.orderingPanelvisible = false;
-    };
-
-    private updateAll() {
-        this.updateServiceClasses();
-        this.updateImageStreams();
-    }
-
-    private updateState() {
-        this.ctrl.loading = ((_.isEmpty(this.ctrl.serviceClasses) && !this.serviceClassesLoaded) ||
-                             (_.isEmpty(this.ctrl.imageStreams) && !this.imageStreamsLoaded));
-
-        if (!this.ctrl.loading) {
-            this.ctrl.filteredItems = this.ctrl.allItems;
-            this.ctrl.categories = this.catalog.removeEmptyCategories(this.ctrl.filteredItems);
-            this.ctrl.subCategories = this.getSubCategories('all');
-        }
-    }
-
-    private updateServiceClasses() {
-        this.ctrl.allItems = this.ctrl.allItems.concat(this.normalizeData('service', this.ctrl.serviceClasses));
-        this.updateState();
+    this.ctrl.currentFilter = category;
+    this.ctrl.currentSubFilter = subCategory || 'all';
+    this.positionExpansionCard();
+  }
+
+  public toggleExpand(subCategory: string) {
+    if (this.ctrl.currentSubFilter === subCategory) {
+      this.ctrl.currentSubFilter = null;
+      this.positionExpansionCard();
+    } else {
+      this.filterByCategory(this.ctrl.currentFilter, subCategory, false);
     }
-
-    private updateImageStreams() {
-        this.ctrl.allItems = this.ctrl.allItems.concat(this.normalizeData('image', this.ctrl.imageStreams));
-        this.updateState();
-    }
-
-    private makeRows(subCats: any) {
-        let subCatsRows: any = [];
-        for (let i = 0, len = subCats.length; i < len; i += this.subCatMaxRowLength) {
-            subCatsRows.push(subCats.slice(i, i + this.subCatMaxRowLength));
-        }
-        return subCatsRows;
-    }
-
-    private getRowOfSubCategory(subCategory: string) {
-        for (let row = 0; row < this.ctrl.subCategories.length; row += 1) {
-            for (let card = 0; card < this.ctrl.subCategories[row].length; card += 1) {
-                if (this.ctrl.subCategories[row][card].id === subCategory) {
-                    return row;
-                }
-            }
-        }
-
-        return 0;
+  }
+
+  public getSubCategories(category: string) {
+    let subCats: any = (category !== 'other') ? [{id: 'all', label:  'All'}] : [];
+    this.ctrl.categories.map(categoryObj => {
+      if (category === 'all' || category === categoryObj.id) {
+        subCats = subCats.concat(categoryObj.subCategories);
+      }
+    });
+    return subCats;
+  };
+
+  public positionExpansionCard() {
+    // set the top position of the expansion card
+    // do in next cycle, after .active card set
+    this.$timeout(() => {
```

Comment:
We should understand why the delay is needed. I'd rather not set it to an arbitrary value. If my laptop is slower then yours, will the timeout fire too soon?

It's better to only call `$timeout` the times we know we need to and leave it off for resize events.